src/Features/Lsif/Generator/Generator.cs

```diff
@@ -196,8 +199,14 @@ public void GenerateForCompilation(Compilation compilation, string projectPath,
             }
 
             lsifJsonWriter.Write(Edge.Create("contains", documentVertex.GetId(), rangeVertices, idFactory));
-            lsifJsonWriter.Write(new Event(Event.EventKind.End, documentVertex.GetId(), idFactory));
 
+            // Write the folding ranges for the document.
+            var foldingRanges = FoldingRangesHandler.GetFoldingRanges(syntaxTree, languageServices, options, isMetadataAsSource: false, CancellationToken.None);
```

Comment:
Out of curiosity, we're passing in an OptionSet to this -- which folding range providers used options in some way?

This code is fine -- what else are you going to do? -- but it's an interesting thing to think about both from the LSIF perspective but also LSP perspective of whether we need to compute things and include extra information that lets the consumer/client know how to interpret it given user options at that point in time.

Reply:
@jasonmalinowski I know that the block structure service itself is using these options: https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/Structure/BlockStructureOptions.cs. I did not check specific options used by the providers.